c3.php

```diff
@@ -8,420 +8,463 @@
+                require_once $this->autoloadRootDir . '/vendor/autoload.php';
+                // Required to load some methods only available at codeception/autoload.php
+                if (stream_resolve_include_path($this->autoloadRootDir . '/vendor/codeception/codeception/autoload.php')) {
+                    require_once $this->autoloadRootDir . '/vendor/codeception/codeception/autoload.php';
+                }
+            } elseif (stream_resolve_include_path('Codeception/autoload.php')) {
+                require_once 'Codeception/autoload.php';
+            } else {
+                __c3_error('Codeception is not loaded. Please check that either PHAR or Composer package can be used');
+            }
+        }
 
-    define('C3_CODECOVERAGE_MEDIATE_STORAGE', Codeception\Configuration::outputDir() . 'c3tmp');
-    define('C3_CODECOVERAGE_PROJECT_ROOT', Codeception\Configuration::projectDir());
-    define('C3_CODECOVERAGE_TESTNAME', $_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE']);
+        // phpunit codecoverage shimming
+        if (!class_exists('PHP_CodeCoverage') and class_exists('SebastianBergmann\CodeCoverage\CodeCoverage')) {
+            class_alias('SebastianBergmann\CodeCoverage\CodeCoverage', 'PHP_CodeCoverage');
+            class_alias('SebastianBergmann\CodeCoverage\Report\Text', 'PHP_CodeCoverage_Report_Text');
+            class_alias('SebastianBergmann\CodeCoverage\Report\PHP', 'PHP_CodeCoverage_Report_PHP');
+            class_alias('SebastianBergmann\CodeCoverage\Report\Clover', 'PHP_CodeCoverage_Report_Clover');
+            class_alias('SebastianBergmann\CodeCoverage\Report\Crap4j', 'PHP_CodeCoverage_Report_Crap4j');
+            class_alias('SebastianBergmann\CodeCoverage\Report\Html\Facade', 'PHP_CodeCoverage_Report_HTML');
+            class_alias('SebastianBergmann\CodeCoverage\Report\Xml\Facade', 'PHP_CodeCoverage_Report_XML');
+            class_alias('SebastianBergmann\CodeCoverage\Exception', 'PHP_CodeCoverage_Exception');
+        }
+        // phpunit version
+        if (!class_exists('PHPUnit_Runner_Version') && class_exists('PHPUnit\Runner\Version')) {
+            class_alias('PHPUnit\Runner\Version', 'PHPUnit_Runner_Version');
+        }
 
-    function __c3_build_html_report(PHP_CodeCoverage $codeCoverage, $path)
-    {
-        $writer = new PHP_CodeCoverage_Report_HTML();
-        $writer->process($codeCoverage, $path . 'html');
+        // Load Codeception Config
+        $configDistFile = $this->configDir . DIRECTORY_SEPARATOR . 'codeception.dist.yml';
+        $configFile = $this->configDir . DIRECTORY_SEPARATOR . 'codeception.yml';
 
-        if (file_exists($path . '.tar')) {
-            unlink($path . '.tar');
+        if (isset($_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'])) {
+            $configFile = $this->configDir . DIRECTORY_SEPARATOR . $_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_CONFIG'];
+        }
```

Comment:
I'm personally not a fan of having this artifact in the code. We should not be dynamically loading user supplied paths.
I imagine this is a very niche cache where you want to dynamically configure it for C3 based on the request.